packages/cardpay-sdk/sdk/revenue-pool/base.ts

```diff
@@ -171,18 +184,36 @@ export default class RevenuePool {
       ZERO_ADDRESS
     );
 
+    let txnHash = gnosisResult.ethereumTx.txHash;
     if (typeof onTxnHash === 'function') {
-      await onTxnHash(gnosisResult.ethereumTx.txHash);
+      await onTxnHash(txnHash);
     }
-    return await waitUntilTransactionMined(this.layer2Web3, gnosisResult.ethereumTx.txHash);
+    return await waitUntilTransactionMined(this.layer2Web3, txnHash);
   }
 
+  async registerMerchant(txnHash: string): Promise<{ merchantSafe: MerchantSafe; txReceipt: TransactionReceipt }>;
   async registerMerchant(
     prepaidCardAddress: string,
-    infoDID: string | undefined,
+    infoDID?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<{ merchantSafe: MerchantSafe; txReceipt: TransactionReceipt }>;
+  async registerMerchant(
+    prepaidCardAddressOrTxnHash: string,
+    infoDID?: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
   ): Promise<{ merchantSafe: MerchantSafe; txReceipt: TransactionReceipt }> {
+    if (!prepaidCardAddressOrTxnHash.startsWith('0x')) {
```

Comment:
I don't understand this test, all hex values start with `0x` and both a prepaid card address and a txn hash are hex values and start with `0x`. Why don't you test to see whether this value looks like an address or a txnHash?